src/shogun/machine/Machine.h

```diff
@@ -116,6 +51,12 @@ enum EProblemType
 	PT_CLASS = 5
 };
 
+/** SVM properties */
+enum SVMProperties : uint64_t
```

Comment:
the problem with this design is that actually this encourages that there are as many enums (and values) as many you like... this will cause problems when we have collision in bit values across enums and then the supports will report false values....

Reply:
Should we "namespace" them. For example, any general machine attribute will have specific number of bits from 0 to m. Any svm machine can take from m to n, and so on.